src/Microsoft.DotNet.Build.Tasks/PackageFiles/BuildVersion.targets

```diff
@@ -0,0 +1,15 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <PropertyGroup>
+    <TodayTimeStamp>$([System.DateTime]::Now.ToString(yyyyMMdd))</TodayTimeStamp>
+    <BuildVersionFile Condition="'$(BuildVersionFile)'==''">$(BaseIntermediateOutputPath)/$(TodayTimeStamp)/BuildVersion.props</BuildVersionFile>
+  </PropertyGroup>
+
+  <!-- If BuildValues.props exists already then import it to get BuildNumberMajor, else generate it and override props values. -->
+  <Import Condition="Exists('$(BuildVersionFile)')" Project="$(BuildVersionFile)" />
+
+  <PropertyGroup Condition="!Exists('$(BuildVersionFile)')">
```

Comment:
This isn't a very good way to decide whether or not to generate this file or not. As you will always get trying to do it in all the initial set of projects you build. Instead we should have inputs and outputs on the target that generates the file itself.